src/libraries/System.Memory/tests/ArrayBufferWriter/ArrayBufferWriterTests.Byte.cs

```diff
@@ -100,8 +100,14 @@ public void GetMemory_ExceedMaximumBufferSize()
         {
             var output = new ArrayBufferWriter<byte>(int.MaxValue / 2 + 1);
             output.Advance(int.MaxValue / 2 + 1);
-            Memory<byte> memory = output.GetMemory(1); // Validate we can't double the buffer size, but can grow by sizeHint
-            Assert.Equal(1, memory.Length);
+
+            // Validate we can't double the buffer size, but can grow
+            Memory<byte> memory = output.GetMemory(1);
+
+            // The buffer should grow more than the 1 byte requested otherwise performance will not
+            // be usable between 1GB and 2GB
+            Assert.True(memory.Length > 1);
```

Comment:
The contract of GetMemory and the IBW interface says that at least the required size is returned. I understand we want to be able to return more, but I a not sure if we should encode that in the test like this.
```suggestion
            Assert.True(memory.Length >= 1);
```

If we truly want to ensure that this particular implementation of the IBW remains fast beyond 1GB, then let's make a stronger/precise, implementation specific assert, something like:
```suggestion
            Assert.Equal(<some number like 256 MB>, memory.Length);
```

Reply:
```suggestion
            Assert.Equal(0X7FEFFFFF, memory.Length);
```

Reply:
Adding `==1` was implied  by the contract, so that's why it was `>1`.

Changing to the magic number makes sense. It couples the test tightly to the implementation which I didn't want to do, but since the test was already coupled previously using it as a precedent it should be fine.